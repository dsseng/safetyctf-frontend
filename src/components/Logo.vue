<template>
  <svg :width='width' :height='height' :viewBox="isVertical ? '0,0,215,175':'0,0,450,115'">
    <g>
      <rect x='55' y='5' width='100' height='100' fill='#138971'></rect>
      <rect x='60' y='10' width='90' height='90' fill='#22baa0'></rect>
      <rect x='75' y='25' width='60' height='60' fill='#138971'></rect>
      <text x='90' y='70' fill='white' stroke='white' style='font: 45px Righteous'>S</text>
    </g>
    <text v-if='isVertical' x='5' y='160' fill='#138971' stroke='#138971' style='font: 45px Roboto'>SafetyCTF</text>
    <text v-else x='175' y='75' fill='#138971' stroke='#138971' style='font: 60px Roboto'>SafetyCTF</text>
  </svg>
</template>

<script>
export default {
  name: 'Logo',
  props: ['width', 'height', 'isVertical']
}
</script>
