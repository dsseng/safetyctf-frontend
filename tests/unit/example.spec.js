/*
import { shallowMount } from "@vue/test-utils";
import Logo from "@/components/Logo.vue";
*/

const { shallowMount } = require("@vue/test-utils");
const Logo = require("@/components/Logo");

describe("Logo.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(Logo, {
      propsData: {}
    });
    expect(wrapper.$el).toMatchSnapshot();
  });

  it("renders correctly if is vertical", () => {
    const wrapper = shallowMount(Logo, {
      propsData: {
        isVertical: true
      }
    });
    expect(wrapper.$el).toMatchSnapshot();
  });

  it("renders correctly with custom width and height", () => {
    const wrapper = shallowMount(Logo, {
      propsData: {
        width: 900,
        height: 230
      }
    });
    expect(wrapper.$el).toMatchSnapshot();
  });
});
