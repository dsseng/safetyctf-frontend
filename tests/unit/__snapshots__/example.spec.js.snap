// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo.vue renders correctly 1`] = `undefined`;

exports[`Logo.vue renders correctly if is vertical 1`] = `undefined`;

exports[`Logo.vue renders correctly with custom width and height 1`] = `undefined`;
